src/parser/jobs/blm/BlmCooldowns.js

```diff
@@ -0,0 +1,148 @@
+//import React, {Fragment} from 'react'
+//import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS, {getAction} from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+//import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+
+//tracking the important™ CDs
+const TRACKEDCDS = [
+ACTIONS.LEY_LINES.id,
+ACTIONS.SHARPCAST.id,
+ACTIONS.TRIPLECAST.id,
+ACTIONS.CONVERT.id,
+]
+
+//Time that laqi deems ok for a OGCD to be down : ^)
+const DOWNTIME_OK_TIME = 15000
+
+export default class Blmcooldowns extends Module {
+	static handle = 'blmcooldowns'
+	static dependencies = [
+		'combatants',
+		'cooldowns',
+		'suggestions',
+		'gauge',
+		'checklist',
+	]
+
+	_downTime = {}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('begincast', {by: 'player'}, this._onBegin)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onBegin(event) {
+
+		//do the check on begin (even tho they don't have a cast time)
+		this._cooldownCheck(event)
+	}
+
+	_onCast(event) {
+
+		//do the check at every cast
+		this._cooldownCheck(event)
+	}
+
+	_onComplete(event) {
+		const endTime = event.timestamp
+		const startTime = this.parser.fight.start_time
+		const encounterLength = endTime - startTime
+		const OGCDRequirements = []
+		Object.keys(this._downTime).forEach(id => {
+			const dt = this._downTime[id]
+			
+			//wrap up all the open ones and save them in history
+			if (dt.current) {
+				dt.current.stoptime = endTime
+				dt.history.push(dt.current)
+				dt.current = null
+			}
+
+			//calculate the downtime based on the start and stop values and sum the array
+			const totalSumOfDownTime = dt.history.map(downTime => {
+				return Math.max(downTime.stoptime - downTime.starttime,0)
+			}).reduce(
+				(accumulator, currentValue) => accumulator + currentValue
+			)
+
+			//write the results as a new Requirement to show up later
+			OGCDRequirements.push(
+				new Requirement({
+				name: getAction(id).name,
+				percent: this._percentFunction(id, totalSumOfDownTime, encounterLength)
+				})
+			)
+		})
+
+		//new Rule and adds the array of Requirements that just got generated
+		this.checklist.add(new Rule({
+			name: 'Use your OGCDs',
+			description: 'Always make sure to use your OGCDs when they are up but don\'t clip them. Utilize your procs or fast Blizzard III or Fire IIIs.',
+			requirements: OGCDRequirements,
+		}))
+	}
+
+	//cool function that I invented that just sets ok usage as 98% and falls very quickly

```

Comment:
While I understand what you're trying to do with this, the current setup doesn't lend itself to being very clear.

Someone looks at a suggestion and they see a really low score when they know they used it a _reasonable_ amount, and will think it's bugging out - possibly even ignore it.

Reply:
well a reasonable ammount is 98% even slightly below a reasonable amount is 95%. I didn't want to use the percent at all but no idea how to make something different with Rule.js